Conflict between discord.js + @nodes/types after upgrading

### Which package is this bug report for?

discord.js

### Issue description

Hi there, approximately two hours ago @nodes/types upgraded their package. I use GitHub Actions to build and test my discord bot, and I didn't realize I was out-of-date until I got conflicting errors that were very confusing.

As far as I can tell, I believe that @ nodes/types did some important changes and now discord.js is conflicting with those changes, and this is an issue with discord.js but I'm not sure. I'm not overly a huge expert on these types of things.

You can replicate this issue by installing the latest @nodes/types and the latest development build of discord.js. Upon attempting to build your project, you will be greeted with this error:

```
Error: node_modules/@types/node/events.d.ts(519,30): error TS2300: Duplicate identifier 'EventEmitter'.
Error: node_modules/@types/node/stream.d.ts(53,11): error TS2420: Class 'ReadableBase' incorrectly implements interface 'ReadableStream'.
  Type 'ReadableBase' is missing the following properties from type 'ReadableStream': off, removeAllListeners, setMaxListeners, getMaxListeners, and 4 more.
Error: node_modules/@types/node/stream.d.ts(662,11): error TS2420: Class 'WritableBase' incorrectly implements interface 'WritableStream'.
  Type 'WritableBase' is missing the following properties from type 'WritableStream': off, removeAllListeners, setMaxListeners, getMaxListeners, and 4 more.
Error: node_modules/@types/node/test.d.ts(380,11): error TS2420: Class 'TestsStream' incorrectly implements interface 'ReadableStream'.
  Type 'TestsStream' is missing the following properties from type 'ReadableStream': off, removeAllListeners, setMaxListeners, getMaxListeners, and 4 more.
Error: node_modules/discord.js/typings/index.d.ts(246,[9](https://github.com/Polytoria/CommunityBot/actions/runs/9605412040/job/26492987136#step:6:10)): error TS2300: Duplicate identifier 'EventEmitter'.
Error: Process completed with exit code 2.
```

### Code sample

```typescript
// <reference path="index.d.ts"/>
import { Client, GatewayIntentBits, ActivityType, Events, Collection, BaseInteraction } from 'discord.js'
import dotenv from 'dotenv'
import { success, warning } from './utils/log.js'
import commandsData from './commandsData.js'
import { IConfiguration } from '../types'

// Initialize .env file.
dotenv.config()

const configuration: IConfiguration = {
  token: process.env.TOKEN,
  clientId: process.env.CLIENTID,
  coolDown: 3
}

const client = new Client({
  intents: [
    GatewayIntentBits.GuildMessages,
    GatewayIntentBits.Guilds,
    GatewayIntentBits.GuildMembers,
    GatewayIntentBits.MessageContent
  ]
})

// @ts-expect-error
client.commands = new Collection()

commandsData.forEach((commandData, index) => {
  // @ts-expect-error
  client.commands.set(commandData.data.name, commandData)
})

client.on('ready', () => {
  // @ts-expect-error
  client.user.setActivity(`${client.guilds.cache.reduce((a, guild) => a + guild.memberCount, 0)} Users`, { type: ActivityType.Watching })

  setInterval(function () {
    // @ts-expect-error
    client.user.setActivity(`${client.guilds.cache.reduce((a, guild) => a + guild.memberCount, 0)} Users`, { type: ActivityType.Watching })
  }, 60000)
})
success({ context: '[Bot]', message: 'Bot succesfully started.' })

client.on('messageCreate', async (message) => {
  if (message.author.bot) return
  if (!message.content.startsWith('p!')) return
  if (!message.inGuild) return

  await message.reply('The Polytoria Community Bot has switched to slash commands!')
})

client.on(Events.InteractionCreate, async (interaction:BaseInteraction) => {
  if (!interaction.isCommand()) {
    return
  }

  // @ts-expect-error
  const command:any = interaction.client.commands.get(interaction.commandName)

  if (!command) {
    interaction.reply("Command doesn't exist")
    return
  }

  success({
    context: '[Client]',
    message: 'Command registered.'
  })

  success({
    context: '[Bot]',
    message: 'Running command ' + command.data.name
  })

  try {
    if (command.constructor.name === 'AsyncFunction') {
      await command.execute(interaction)
    } else {
      command.execute(interaction)
    }
  } catch (error: any) {
    if (interaction.replied) {
      await interaction.followUp('Failed to execute command: ' + error)
    } else {
      await interaction.reply('Failed to execute command: ' + error)
    }
    warning({
      context: '[Bot]',
      message: error.toString()
    })
  }
})

// Handle Promise Rejection
process.on('unhandledRejection', (reason, p) => {
  console.error(reason, 'Unhandled Rejection at Promise', p)
})

process.on('uncaughtException', err => {
  console.error(err)
  process.exit(1)
})

success({ context: '[Bot]', message: 'Bot succesfully logged in.' })

client.login(configuration.token)
```


### Versions

typescript - 5.5.2
discord-js - latest development build
@types/node - 20.14.7

Node - Latest v20

### Issue priority

High (immediate attention needed)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

Guilds, GuildMembers, GuildMessages, MessageContent

### I have tested this issue on a development release

_No response_

--- a/packages/discord.js/typings/index.d.ts
+++ b/packages/discord.js/typings/index.d.ts
@@ -243,20 +243,6 @@ import {
   RawWidgetMemberData,
 } from './rawDataTypes.js';
 
-declare module 'node:events' {
-  class EventEmitter {
-    // Add type overloads for client events.
-    public static once<Emitter extends EventEmitter, Event extends keyof ClientEvents>(
-      eventEmitter: Emitter,
-      eventName: Emitter extends Client ? Event : string,
-    ): Promise<Emitter extends Client ? ClientEvents[Event] : any[]>;
-    public static on<Emitter extends EventEmitter, Events extends keyof ClientEvents>(
-      eventEmitter: Emitter,
-      eventName: Emitter extends Client ? Events : string,
-    ): AsyncIterableIterator<Emitter extends Client ? ClientEvents[Events] : any>;
-  }
-}
-
 //#region Classes
 
 export class Activity {
@@ -996,6 +982,18 @@ export class Client<Ready extends boolean = boolean> extends BaseClient {
   private get _censoredToken(): string | null;
   // This a technique used to brand the ready state. Or else we'll get `never` errors on typeguard checks.
   private readonly _ready: Ready;
+
+  // Override inherited static EventEmitter methods, with added type checks for Client events.
+  public static once<Emitter extends EventEmitter, Event extends keyof ClientEvents>(
+    eventEmitter: Emitter,
+    eventName: Emitter extends Client ? Event : string | symbol,
+    options?: { signal?: AbortSignal | undefined },
+  ): Promise<Emitter extends Client ? ClientEvents[Event] : any[]>;
+  public static on<Emitter extends EventEmitter, Event extends keyof ClientEvents>(
+    eventEmitter: Emitter,
+    eventName: Emitter extends Client ? Event : string | symbol,
+    options?: { signal?: AbortSignal | undefined },
+  ): AsyncIterableIterator<Emitter extends Client ? ClientEvents[Event] : any[]>;
 
   public application: If<Ready, ClientApplication>;
   public channels: ChannelManager;

--- a/packages/discord.js/typings/index.test-d.ts
+++ b/packages/discord.js/typings/index.test-d.ts
@@ -1294,6 +1294,10 @@ client.on('guildCreate', async g => {
   );
 });
 
+// EventEmitter static method overrides
+expectType<Promise<[Client<true>]>>(Client.once(client, 'ready'));
+expectType<AsyncIterableIterator<[Client<true>]>>(Client.on(client, 'ready'));
+
 client.login('absolutely-valid-token');
 
 declare const loggedInClient: Client<true>;

